src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -160,10 +161,21 @@ public void setup(
     }
 
     public void start(Federation federation) {
+       FederationMember federationMember = federatorSupport.getFederationMember();
+       if (federation.isMember(federationMember)) {
+            String message = String.format(
+                "Member %s is no part of the federation %s",
+                federationMember.getBtcPublicKey(),
+                federation.getAddress());
+            logger.error("[start] {}", message);
+            throw new IllegalStateException(message);
+        }
+
         if (!observedFederations.contains(federation)) {
             observedFederations.add(federation);
-            logger.debug("[start] observing Federation {}", federation.getAddress());
+            logger.debug("[start] Observing federation {}", federation.getAddress());
         }
+
         if (observedFederations.size() == 1) {
             // If there is just one observed Federation, it means the btcReleaseClient wasn't started
             logger.debug("[start] Starting");
```

Comment:
Since the listener just gets activated once there is one observed federation available. Consider the case you start it again with a different federation. Then you will have two listener instances running in the background doing the same computation twice. Thus, when size is 1 it means the listener for the RSK blockchain gets activated, if it decreases to zero then there is no need for the listener to be running.